File: src/main/java/Input.java
import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.File;
import java.io.IOException;

public class Input {


    private String semester;
    private int courseFFRate;
    private int quotaForElectives;
    private int quotaForMandatory;
    private int maxNumberOfSelectionForCourses;



    private String coursesJsonName;
    private String electiveNTEJsonFileName;
    private String electiveFTEJsonFileName;
    private String electiveUEJsonFileName;
    private String electiveTEJsonFileName;
    private String advisorsJsonName;
    private String studentsJsonName;





    private Courses[] courses;
    private Person[] students;
    private Person[] advisors;
    private Courses[] NTE;
    private Courses[] FTE;
    private Courses[] UE;
    private Courses[] TE;



    public void setSemester(String semester) {
        this.semester = semester;
    }

    public int getCourseFFRate() {
        return courseFFRate;
    }

    public void setCourseFFRate(int courseFFRate) {
        this.courseFFRate = courseFFRate;
    }

    public int getQuotaForElectives() {
        return quotaForElectives;
    }

    public void setQuotaForElectives(int quotaForElectives) {
        this.quotaForElectives = quotaForElectives;
    }

    public int getQuotaForMandatory() {
        return quotaForMandatory;
    }

    public void setQuotaForMandatory(int quotaForMandatory) {
        this.quotaForMandatory = quotaForMandatory;
    }

    public int getMaxNumberOfSelectionForCourses() {
        return maxNumberOfSelectionForCourses;
    }

    public void setMaxNumberOfSelectionForCourses(int maxNumberOfSelectonForCourses) {
        this.maxNumberOfSelectionForCourses = maxNumberOfSelectonForCourses;
    }

    public void setCoursesJsonName(String coursesJsonName) {
        this.coursesJsonName = coursesJsonName;
    }

    public void setElectiveNTEJsonFileName(String electiveNTEJsonFileName) {
        this.electiveNTEJsonFileName = electiveNTEJsonFileName;
    }

    public void setElectiveFTEJsonFileName(String electiveFTEJsonFileName) {
        this.electiveFTEJsonFileName = electiveFTEJsonFileName;
    }

    public void setElectiveUEJsonFileName(String electiveUEJsonFileName) {
        this.electiveUEJsonFileName = electiveUEJsonFileName;
    }

    public void setElectiveTEJsonFileName(String electiveTEJsonFileName) {
        this.electiveTEJsonFileName = electiveTEJsonFileName;
    }

    public void setAdvisorsJsonName(String advisorsJsonName) {
        this.advisorsJsonName = advisorsJsonName;
    }

    public void setStudentsJsonName(String studentsJsonName) {
        this.studentsJsonName = studentsJsonName;
    }

    public void createObjects() throws IOException {
        ObjectMapper objectMapper = new ObjectMapper();

        File advisorJsonFile = new File(advisorsJsonName);
        Person[] advisors = objectMapper.readValue(advisorJsonFile, Advisor[].class);

        File lecturesJsonFile = new File(coursesJsonName);
        Courses[] courses = objectMapper.readValue(lecturesJsonFile, Courses[].class);

        File UEJson = new File(electiveUEJsonFileName);
        Courses[] UE = objectMapper.readValue(UEJson, Courses[].class);

        File TEJson = new File(electiveTEJsonFileName);
        Courses[] TE = objectMapper.readValue(TEJson, Courses[].class);

        File NTEJson = new File(electiveNTEJsonFileName);
        Courses[] NTE = objectMapper.readValue(NTEJson, Courses[].class);

        File FTEJson = new File(electiveFTEJsonFileName);
        Courses[] FTE = objectMapper.readValue(FTEJson, Courses[].class);


        File studentsJsonFile = new File(studentsJsonName);
        Person[] students = objectMapper.readValue(studentsJsonFile, Student[].class);

        this.advisors = advisors;
        this.courses = courses;
        this.UE = UE;
        this.TE = TE;
        this.NTE = NTE;
        this.FTE = FTE;
        this.students = students;

    }

    public void startSimulationWithInputs() throws IOException, IllegalAccessException {
        GenerateStudent generateStudent = new GenerateStudent((Student[]) students, courses, UE, TE, NTE, FTE, (Advisor[]) advisors);
        generateStudent.simulate();
    }
}

File: input.py
import json
from courses import Courses
from student import Student
from advisor import Advisor
from generateStudent import GenerateStudent

class Input:
    def __init__(self, data):
        self.__courses = []
        self.__nte = []
        self.__fte = []
        self.__ue = []
        self.__te = []
        self.__advisors = []
        self.__students = []
        self.__semester = data['semester']
        self.__course_ff_rate = data['courseFFRate']
        self.__quota_for_electives = data['quotaForElectives']
        self.__quota_for_mandatory = data['quotaForMandatory']
        self.__max_number_of_selection_for_courses = data['maxNumberOfSelectionForCourses']
        self.__courses_json_name = data['coursesJsonName']
        self.__elective_nte_json_file_name = data['electiveNTEJsonFileName']
        self.__elective_fte_json_file_name = data['electiveFTEJsonFileName']
        self.__elective_ue_json_file_name = data['electiveUEJsonFileName']
        self.__elective_te_json_file_name = data['electiveTEJsonFileName']
        self.__advisors_json_name = data['advisorsJsonName']
        self.__students_json_name = data['studentsJsonName']

    def set_quota(self):
        length = len(self.__courses)
        for i in range(length):
            self.__courses[i].quota = self.__quota_for_mandatory
        length = len(self.__nte)
        for i in range(length):
            self.__nte[i].quota = self.__quota_for_electives
        length = len(self.__fte)
        for i in range(length):
            self.__fte[i].quota = self.__quota_for_electives
        length = len(self.__ue)
        for i in range(length):
            self.__ue[i].quota = self.__quota_for_electives
        length = len(self.__te)
        for i in range(length):
            self.__te[i].quota = self.__quota_for_electives

    def create_courses(self):
        with open(self.__courses_json_name, 'r') as f:
            courses_data = json.load(f)
        for data in courses_data:
            temp_course = Courses(data)
            self.__courses.append(temp_course)

        with open(self.__elective_nte_json_file_name, 'r') as f:
            nte_data = json.load(f)
        for data in nte_data:
            temp_course = Courses(data)
            self.__nte.append(temp_course)

        with open(self.__elective_fte_json_file_name, 'r') as f:
            fte_data = json.load(f)
        for data in fte_data:
            temp_course = Courses(data)
            self.__fte.append(temp_course)

        with open(self.__elective_ue_json_file_name, 'r') as f:
            ue_data = json.load(f)
        for data in ue_data:
            temp_course = Courses(data)
            self.__ue.append(temp_course)

        with open(self.__elective_te_json_file_name, 'r') as f:
            te_data = json.load(f)
        for data in te_data:
            temp_course = Courses(data)
            self.__te.append(temp_course)
        self.set_quota()

    def create_objects(self):
        self.create_student()
        self.create_advisor()
        self.create_courses()

    def create_advisor(self):
        with open(self.__advisors_json_name, 'r') as f:
            advisor_data = json.load(f)
        for data in advisor_data:
            temp_advisor = Advisor(data)
            self.__advisors.append(temp_advisor)

    def create_student(self):
        with open(self.__students_json_name, 'r') as f:
            student_data = json.load(f)
        for data in student_data:
            temp_student = Student(data)
            self.__students.append(temp_student)

    def start_simulation_with_inputs(self):
        generate_student = GenerateStudent(self.__students, self.__courses, self.__ue, self.__te, self.__nte, self.__fte, self.__advisors, self.__course_ff_rate, self.__max_number_of_selection_for_courses, self.__semester)
        generate_student.simulate()